server/src/main/java/org/elasticsearch/index/codec/vectors/OptimizedScalarQuantizer.java

```diff
@@ -141,6 +141,36 @@ public QuantizationResult scalarQuantize(float[] vector, byte[] destination, byt
         );
     }
 
+    public QuantizationResult scalarQuantizeToInts(float[] vector, int[] destination, byte bits, float[] centroid) {
```

Comment:
if we are confident in the speed improvements, I would argue that we shouldn't bother with this new method, and simply adjust the APIs.

The on disk format is unchanged, so any older formats should be able to be adjusted that rely on the `scalarQuantize` signatures.